cores/esp8266/Schedule.cpp

```diff
@@ -122,6 +123,15 @@ bool schedule_recurrent_function_us(const std::function<bool(void)>& fn,
 
     esp8266::InterruptLock lockAllInterruptsInThisScope;
 
+    // prevent new item overwriting an already expired rTarget.
+    const auto now = micros();
+    const auto itemRemaining = item->callNow.remaining();
+    const int32_t remaining = rTarget - now;
+    if (!rFirst || (remaining > 0 && static_cast<decltype(micros())>(remaining) > itemRemaining))
+    {
+        rTarget = now + itemRemaining;
+    }
+
```

Comment:
Without needing `remaining` and `static_cast`:
```
if (!rFirst || (rTarget > now && rTarget > now + itemRemaining))
```
?

Reply:
I have changed the remaining int32_t uses and the respective casts.